STM32Cube/Tasks/vn_handler.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * vn_handler.c
+ *
+ *  Created on: Mar 24, 2024
+ *      Author: joedo
+ */
+
+
+#include "vn_handler.h"
+#include "vn/protocol/upack.h"
+#include "FreeRTOS.h"
+#include "semphr.h"
+#include "fake_logging.h"
+#include "stm32h7xx_hal.h"
+
+
+//RateDivisor sets the output rate as a function of the sensor ImuRate (800 Hz for the VN-200)
+#define CONFIG_MSG_LENGTH 100
+#define VN_200_IMURATE 800 //Hz
+
+//Binary Output #1
+#define BINARY1_OUT_RATE 10 //Hz
+#define RATE1_DIVISOR VN_200_IMURATE / BINARY1_OUT_RATE
+
+//See "vnenum.h" and the ICD for parameter definitions
+/*
+#define COMMONGROUP_PARAMS COMMONGROUP_QUATERNION | COMMONGROUP_POSITION | COMMONGROUP_VELOCITY | COMMONGROUP_ACCEL
+#define TIMEGROUP_PARAMS TIMEGROUP_TIMEUTC | TIMEGROUP_TIMEUTC
+#define IMUGROUP_PARAMS IMUGROUP_IMUSTATUS | IMUGROUP_SENSSAT | IMUGROUP_UNCOMPMAG | IMUGROUP_UNCOMPACCEL | IMUGROUP_UNCOMPGYRO | IMUGROUP_TEMP | IMUGROUP_PRES
+#define GPSGROUP_PARAMS GPSGROUP_UTC | GPSGROUP_NUMSATS | GPSGROUP_FIX | GPSGROUP_POSLLA | GPSGROUP_VELECEF
+#define ATTITUDEGROUP_PARAMS ATTITUDEGROUP_NONE
+#define INSGROUP_PARAMS INSGROUP_NONE
+*/
+
+//Minimize binary output contents for initial testing
+#define COMMONGROUP_PARAMS COMMONGROUP_NONE
+#define TIMEGROUP_PARAMS TIMEGROUP_TIMEUTC
+#define IMUGROUP_PARAMS IMUGROUP_NONE
+#define GPSGROUP_PARAMS GPSGROUP_NONE
+#define ATTITUDEGROUP_PARAMS ATTITUDEGROUP_NONE
+#define INSGROUP_PARAMS INSGROUP_NONE
+
+
+#define MAX_BINARY_OUTPUT_LENGTH 200
+#define DMA_RX_TIMEOUT 300
+uint8_t USART1_Rx_Buffer[MAX_BINARY_OUTPUT_LENGTH];
+SemaphoreHandle_t USART1_DMA_Sempahore;
+
+void USART1_DMA_Rx_Complete_Callback(UART_HandleTypeDef *huart)
+{
+	BaseType_t xHigherPriorityTaskWoken = pdFALSE;
+	xSemaphoreGiveFromISR(USART1_DMA_Sempahore, &xHigherPriorityTaskWoken);
+	portYIELD_FROM_ISR( xHigherPriorityTaskWoken );
+}
+
+int vnIMUSetup(){
```

Comment:
Might be clearer to return either HAL_StatusTypeDef, or bool 0/1.

Reply:
yeah I agree. We can use the bool convention since that is what most RocketCAN/canlib stuff uses.